### ISSUE TITLE
Support using org.jooq.impl.EnumConverter as a <converter> configuration

### ISSUE THREAD

**Issue by yissachar** [jOOQ/jOOQ#6567]
### Expected behavior and actual behavior:
Codegen should generate working code when using custom data types.

### Steps to reproduce the problem:

I'm using the following Gradle config:

```
database {
    name = 'org.jooq.util.postgres.PostgresDatabase'
    inputSchema = 'public'

    customTypes {
        customType {
            name = "EnumThing"
            type = "com.my.custom.EnumThing"
            converter = "org.jooq.impl.EnumConverter"
        }
    }

    forcedTypes {
        forcedType {
            name = "EnumThing"
            expression = "afield"
        }
    }
}
```

The POJO and Record are generated correctly, but the table has the following line:

```
public final TableField<MyTableRecord, EnumThing> AFIELD = createField("afield", org.jooq.impl.SQLDataType.VARCHAR.length(30), this, "", new EnumConverter());
```

The problem is `EnumConverter` does not have a no-arg constructor so this won't compile.

I'm not sure if I'm doing something wrong, but as far as I can tell I am following the example correctly.

### Versions:

- jOOQ: 3.9.3
- Database (include vendor): Postgres 9.6
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for reporting. We could probably generate the appropriate constructor call, as we know both types, the `<T>` type (database type) and the `<U>` type (your enum type) that need to be passed to the `EnumConverter` constructor.

In the meantime, as a workaround: the original intent of jOOQ's `EnumConverter` utility is to subclass it and pass that to your configuration.

Hope this helps

**Comment by lukaseder** [comment]
Related: #5877

**Comment by lukaseder** [comment]
Fixed for the upcoming jOOQ 3.10. Thanks again for your suggestion.

**Comment by yissachar** [comment]
Awesome, thanks for the quick fix! Subclassing works fine for now.
